File: proxy/node/js/helper/timestamps.js
module.exports = {
    addTimestamps,
    getTimestamp
}

function addTimestamps(data, received) {
  const d = JSON.parse(data)

  d.server = {}
  d.server.received = received
  d.server.sent = getTimestamp()

  return d
}

function getTimestamp() {
  return new Date(Date.now()).toISOString()
}

File: proxy/bottle/py/helper/timestamps.py
import json
from datetime import datetime

def getTimestamp():
    return str(datetime.utcnow().isoformat())

def addTimestamps(data, received):
    d = json.loads(data)

    d['server'] = {}
    d['server']['received'] = received
    d['server']['sent'] = getTimestamp()

    return d
